ecosystem/typescript/sdk/examples/typescript/ambassador.ts

```diff
@@ -0,0 +1,153 @@
+
+        return pendingTxn.hash;
+    }
+
+    async burn(creator: AptosAccount, token: HexString): Promise<string> {
+        const rawTxn = await provider.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::burn`,
+            type_arguments: [],
+            arguments: [token.hex()],
+        });
+
+        const bcsTxn = await provider.signTransaction(creator, rawTxn);
+        const pendingTxn = await provider.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async mintAmbassadorToken(creator: AptosAccount, description: string, name: string, uri: string, soul_bound_to: HexString): Promise<string> {
+        const rawTxn = await provider.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::mint_ambassador_token`,
+            type_arguments: [],
+            arguments: [description, name, uri, soul_bound_to.hex()],
+        });
+
+        const bcsTxn = await provider.signTransaction(creator, rawTxn);
+        const pendingTxn = await provider.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async ambassadorLevel(creator_addr: HexString, token_addr: HexString): Promise<bigint> {
+        const payload: Types.ViewRequest = {
+            function: `${creator_addr.hex()}::ambassador::ambassador_level`,
+            type_arguments: [],
+            arguments: [token_addr.hex()],
+        };
+
+        const result = await provider.view(payload);
+        return BigInt(result[0] as any);
+    }
+}
+
+/** run our demo! */
+async function main(): Promise<void> {
+    const client = new AmbassadorClient();
+
+    const admin = new AptosAccount();
+    const user = new AptosAccount();
+
+    await faucetClient.fundAccount(admin.address(), 100_000_000);
+    await faucetClient.fundAccount(user.address(), 100_000_000);
+
+    console.log("\nCompile and publish the Ambassador module (`aptos-core/aptos-move/move-examples/token_objects/ambassador`) using the following profile, and press enter:");
+    console.log(`  ambassador_admin:`);
+    console.log(`    private_key: "${admin.toPrivateKeyObject().privateKeyHex}"`);
+    console.log(`    public_key: "${admin.pubKey()}"`);
+    console.log(`    account: ${admin.address()}`);
+    console.log(`    rest_url: "https://fullnode.testnet.aptoslabs.com"`);
+    console.log(`    faucet_url: "https://faucet.testnet.aptoslabs.com"`);
+
+    await waitForEnter();
+
+    const adminAddr = admin.address()
+    const userAddr = user.address()
+    const tokenName = "Aptos Ambassador #1";
+
+    console.log("\n=== Addresses ===");
+    console.log(`Admin: ${adminAddr} `);
+    console.log(`User: ${userAddr} `);
+
+    // Mint Ambassador Token
+    let txnHash = await client.mintAmbassadorToken(admin, 'Aptos Ambassador Token', tokenName, 'https://raw.githubusercontent.com/aptos-labs/aptos-core/main/aptos-move/move-examples/token_objects/ambassador/metadata/', userAddr);
```

Comment:
Why not also move these images to here in ecosystem/ ?

Reply:
Done.e